yt_dlp/extractor/mashable.py

```diff
@@ -0,0 +1,64 @@
+import json
+import re
+
+from yt_dlp.utils import ExtractorError, try_get
+from .common import InfoExtractor
+
+
+class MashableExtractorIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mashable\.com/video/(.*)'
+    _TESTS = [{
+        'url': 'https://mashable.com/video/why-life-on-venus-is-better-than-mars',
+        'md5': 'f401a6db2b649d9733c7753474dd2a31',
+        'info_dict': {
+            'id': '50319e30-a4ce-084d',
+            'ext': 'mp4',
+            'duration': 257.0,
+            'title': 'MashableExtractor video #50319e30-a4ce-084d',  # generic assigned by code
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'description': 'Leave Mars to the ultra-rich. It’s Venus we should move to one day.',
+        }
+    },
+        {
+            'url': 'https://mashable.com/video/who-took-the-first-selfie-ever',
+            'md5': '4e83f6efc988ffa242fe7254094b9774',
+            'info_dict': {
+                'id': '5cee6e38-f9c8-5ceb',
+                'ext': 'mp4',
+                'duration': 63.0,
+                'title': 'Who took the first selfie ever?',
+                'thumbnail': r're:^https?://.*\.png$',
+                'description': 'Say cheese!',
+            }
+    }]
+
+    def _real_extract(self, url):
+        webpage = self._download_webpage(url, None)
+        pattern = re.compile(r'data: (.*),')
+        matches = pattern.search(webpage)
+        try:
+            video_metadata = json.loads(matches.group(1))
+            pattern = re.compile(r'https:\/\/vdist.aws.mashable.com\/cms/[0-9]{4}\/[0-9]{1,2}\/(.*-.*-.*)\/mp4\/.*')
+            video_id = pattern.search(video_metadata['url']).group(1)
```

Comment:
use self._search_regex/_parse_json

Reply:
Also, improve regexes

Reply:
Will update to _parse_json. For the improvement of regex would you say the .*- parts for the id specifically is what should be updated?

Reply:
1. Avoid using `.`
2. Don't use `*` where it should be `+`
3. Remove unnecessary escaping

Reply:
Updated! For the data_pattern regex, I spent sometime looking for a cleaner solution to grab regex but quickly went down into messy solutions. VERY open to suggestions if you have any, although I think maybe the .* might be ok here with the 'data:' and ',' around the json. Let me know what you think